indexer/services/addresses.go

```diff
@@ -0,0 +1,109 @@
+package services
+
+import (
+	"github.com/ethereum-optimism/optimism/indexer/bindings/legacy/scc"
+	"github.com/ethereum-optimism/optimism/op-bindings/bindings"
+	"github.com/ethereum-optimism/optimism/op-bindings/predeploys"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type AddressManager interface {
+	L1StandardBridge() (common.Address, *bindings.L1StandardBridge)
+	StateCommitmentChain() (common.Address, *scc.StateCommitmentChain)
+	OptimismPortal() (common.Address, *bindings.OptimismPortal)
+}
+
+type LegacyAddresses struct {
+	l1SB     *bindings.L1StandardBridge
+	l1SBAddr common.Address
+	l2SB     *bindings.L2StandardBridge
+	scc      *scc.StateCommitmentChain
+	sccAddr  common.Address
+}
+
+func NewLegacyAddresses(client bind.ContractBackend, addrMgrAddr common.Address) (AddressManager, error) {
+	mgr, err := bindings.NewAddressManager(addrMgrAddr, client)
+	if err != nil {
+		return nil, err
+	}
+
+	l1SBAddr, err := mgr.GetAddress(nil, "Proxy__OVM_L1StandardBridge")
+	if err != nil {
+		return nil, err
+	}
+	sccAddr, err := mgr.GetAddress(nil, "StateCommitmentChain")
+	if err != nil {
+		return nil, err
+	}
+	l1SB, err := bindings.NewL1StandardBridge(l1SBAddr, client)
+	if err != nil {
+		return nil, err
+	}
+	l2SB, err := bindings.NewL2StandardBridge(predeploys.L2StandardBridgeAddr, client)
```

Comment:
There is no getter for the `L2StandardBridge`?

Reply:
Correct, since we know the address in advance.

Reply:
Ah now I understand what you mean - this is fixed.